specification/appconfiguration/AppConfiguration/models.tsp

```diff
@@ -0,0 +1,332 @@
+
+  @doc("Content type field.")
+  content_type: "content_type",
+
+  @doc("Value field.")
+  value: "value",
+
+  @doc("Last modified field.")
+  last_modified: "last_modified",
+
+  @doc("Tags field.")
+  tags: "tags",
+
+  @doc("Locked field.")
+  locked: "locked",
+
+  @doc("Etag field.")
+  etag: "etag",
+
+  string,
+}
+
+@doc("Snapshot fields.")
+union SnapshotFields {
+  @doc("Name field.")
+  name: "name",
+
+  @doc("Status field.")
+  status: "status",
+
+  @doc("Filters field.")
+  filters: "filters",
+
+  @doc("Composition type field.")
+  composition_type: "composition_type",
+
+  @doc("Created field.")
+  created: "created",
+
+  @doc("Expires field.")
+  expires: "expires",
+
+  @doc("Retention period field.")
+  retention_period: "retention_period",
+
+  @doc("Size field.")
+  size: "size",
+
+  @doc("Items count field.")
+  items_count: "items_count",
+
+  @doc("Tags field.")
+  tags: "tags",
+
+  @doc("Etag field.")
+  etag: "etag",
+
+  string,
+}
+
+@doc("Snapshot status.")
+union SnapshotStatus {
+  @doc("Provisioning")
+  provisioning: "provisioning",
+
+  @doc("Ready")
+  ready: "ready",
+
+  @doc("Archived")
+  archived: "archived",
+
+  @doc("Failed")
+  failed: "failed",
+
+  string,
+}
+
+@doc("Composition types.")
+union CompositionType {
+  @doc("The 'key' composition type.")
+  key: "key",
+
+  @doc("The 'key_label' composition type.")
+  key_label: "key_label",
+
+  string,
+}
+
+@doc("Label fields.")
+union LabelFields {
+  @doc("Name field.")
+  name: "name",
+
+  string,
+}
+
+@doc("The result of a list request.")
+@pagedResult
+model KeyListResult {
+  @doc("The collection value.")
+  @items
+  items?: Key[];
+
+  @doc("The URI that can be used to request the next set of paged results.")
+  @nextLink
+  `@nextLink`?: string;
+}
+
+@doc("Keys serve as identifiers for key-values and are used to store and retrieve corresponding values.")
+@resource("keys")
+model Key {
+  @doc("The name of the key.")
+  @key
+  @visibility("read")
+  name: string;
+}
+
+@doc("Azure App Configuration error object.")
+@error
+model Error {
+  @doc("The type of the error.")
+  type?: string;
+
+  @doc("A brief summary of the error.")
+  title?: string;
+
+  @doc("The name of the parameter that resulted in the error.")
+  name?: string;
+
+  @doc("A detailed description of the error.")
+  detail?: string;
+
+  @doc("The HTTP status code that the error maps to.")
+  status?: int32;
+}
+
+@doc("The result of a list request.")
+@pagedResult
+model KeyValueListResult {
+  @doc("The collection value.")
+  @items
+  items?: KeyValue[];
+
+  @doc("An identifier representing the returned state of the resource.")
+  etag?: string;
+
+  @doc("The URI that can be used to request the next set of paged results.")
+  @nextLink
+  `@nextLink`?: string;
+}
+
+@doc("A key-value pair representing application settings.")
+@resource("kv")
+model KeyValue {
+  @key
+  @visibility("read")
+  @doc("The key of the key-value.")
+  key: string;
```

Comment:
Same question here, can the `key` property be required?

Reply:
Yes